lib/sass/script/functions.rb

```diff
@@ -525,18 +529,27 @@ def initialize(environment)
       #   assert_type value, :String
       #   assert_type value, :Number
       # @param value [Sass::Script::Value::Base] A SassScript value
-      # @param type [Symbol] The name of the type the value is expected to be
+      # @param type [Symbol, Array<Symbol>] The name(s) of the type the value is expected to be
       # @param name [String, Symbol, nil] The name of the argument.
       # @raise [ArgumentError] if value is not of the correct type.
       def assert_type(value, type, name = nil)
```

Comment:
I'm not convinced it's worth adding extra complexity and overhead here versus hand-rolling the verification temporarily for `function-reference()`.